src/mlpack/methods/ann/ffn.hpp

```diff
@@ -172,9 +172,10 @@ class FFN
    * @param i Index of points to use for objective function gradient evaluation.
    * @param gradient Matrix to output gradient into.
    */
+  template<typename eT>
   void Gradient(const arma::mat& parameters,
                 const size_t i,
-                arma::mat& gradient);
+                arma::Mat<eT>& gradient);
```

Comment:
Also, I guess the question here was what is the reason to use ``arma::Mat<eT>`` instead of ``arma::mat``.